models/iaf_tum_2000.h

```diff
@@ -0,0 +1,482 @@
+/*
+ *  iaf_tum_2000.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef IAF_TUM_2000_H
+#define IAF_TUM_2000_H
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "recordables_map.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+// clang-format off
+/* BeginUserDocs: neuron, integrate-and-fire, current-based, synapse, short-term plasticity
+
+Short description
++++++++++++++++++
+
+Leaky integrate-and-fire neuron model with exponential PSCs and integrated short-term plasticity synapse
+
+Description
++++++++++++
+
+``iaf_tum_2000`` is a leaky integrate-and-fire neuron model with short-term synaptic
+plasticity and exponential shaped postsynaptic currents (PSCs). In particular,
+``iaf_tum_2000`` implements short-term depression and short-term facilitation
+according to [1]_ by solving Eqs. (3) and (4) from this paper in an exact manner.
+
+``iaf_tum_2000`` differs from `iaf_psc_exp <../models/iaf_psc_exp>` by the addition
+of synaptic state variables :math:`x`, :math:`z` and :math:`u`, which together
+with the membrane potential :math:`V_\text{m}` and synaptic current :math:`I_\text{syn}`
+obey the following dynamics:
+
+.. math::
+
+   \frac{dV_\text{m}}{dt} &= -\frac{V_{\text{m}} - E_\text{L}}{\tau_{\text{m}}} + \frac{I_{\text{syn}} + I_\text{e}}{C_{\text{m}}}
+
+   I_{\text{syn}} &= I_\text{syn,ex} + I_\text{syn,in}
+
+   I_\text{syn,X} &= \sum_{j \in X} w_j y_j
+
+   \frac{dx_j}{dt} &= \frac{z_j}{\tau_{\text{rec}}} - u_j x_j \delta(t - t_j)
+
+   \frac{dy_j}{dt} &= -\frac{y_j}{\tau_{\text{syn}}} + u_j x_j \delta(t - t_j)
+
+   \frac{dz_j}{dt} &= \frac{y_j}{\tau_{\text{syn}}} - \frac{y_j}{\tau_{\text{rec}}}
+
+   \frac{du_j}{dt} &= -\frac{u}{\tau_{\text{fac}}} + U(1 - u) \delta(t - t_j)
+
+
+where :math:`j` indexes either excitatory (:math:`\text{X} = \text{ex}`)
+or inhibitory (:math:`\text{X} = \text{in}`) presynaptic neurons,
+:math:`k` indexes the spike times of neuron :math:`j`, and :math:`d_j`
+is the delay from neuron :math:`j`.
```

Comment:
I am a bit confused about `j` and `X` here. Isn't `j` a numerical index of neurons? How does the `X` for `ex`/`in` then come in here?

Reply:
Hopefully rectified by [0fd87c0](https://github.com/nest/nest-simulator/pull/2971/commits/0fd87c0a58fe58c174fe1eee7ef8946622938c9b).